service/controller/v3/resource/deployment/deployment.go

```diff
@@ -44,34 +45,37 @@ func (r Resource) newDeployment(ctx context.Context, customObject providerv1alph
 		workerNodes = append(workerNodes, n)
 	}
 
-	masterCloudConfig, err := r.cloudConfig.NewMasterCloudConfig(customObject)
+	sc, err := controllercontext.FromContext(ctx)
 	if err != nil {
 		return azureresource.Deployment{}, microerror.Mask(err)
 	}
 
-	workerCloudConfig, err := r.cloudConfig.NewWorkerCloudConfig(customObject)
+	masterCloudConfig, err := sc.CloudConfig.NewMasterCloudConfig(customObject)
+	if err != nil {
+		return azureresource.Deployment{}, microerror.Mask(err)
+	}
+
+	workerCloudConfig, err := sc.CloudConfig.NewWorkerCloudConfig(customObject)
 	if err != nil {
 		return azureresource.Deployment{}, microerror.Mask(err)
 	}
 
 	defaultParams := map[string]interface{}{
-		"clusterID":                     key.ClusterID(customObject),
-		"virtualNetworkName":            key.VnetName(customObject),
-		"virtualNetworkCidr":            key.VnetCIDR(customObject),
-		"calicoSubnetCidr":              key.VnetCalicoSubnetCIDR(customObject),
-		"masterSubnetCidr":              key.VnetMasterSubnetCIDR(customObject),
-		"workerSubnetCidr":              key.VnetWorkerSubnetCIDR(customObject),
-		"masterNodes":                   masterNodes,
-		"workerNodes":                   workerNodes,
-		"dnsZones":                      key.DNSZones(customObject),
-		"hostClusterCidr":               r.azure.HostCluster.CIDR,
-		"hostClusterResourceGroupName":  r.azure.HostCluster.ResourceGroup,
-		"hostClusterVirtualNetworkName": r.azure.HostCluster.VirtualNetwork,
-		"kubernetesAPISecurePort":       key.APISecurePort(customObject),
-		"masterCloudConfigData":         masterCloudConfig,
-		"workerCloudConfigData":         workerCloudConfig,
-		"vmssMSIEnabled":                r.azure.MSI.Enabled,
-		"templatesBaseURI":              baseTemplateURI(r.templateVersion),
+		"clusterID":               key.ClusterID(customObject),
+		"virtualNetworkName":      key.VnetName(customObject),
+		"virtualNetworkCidr":      key.VnetCIDR(customObject),
+		"calicoSubnetCidr":        key.VnetCalicoSubnetCIDR(customObject),
+		"masterSubnetCidr":        key.VnetMasterSubnetCIDR(customObject),
+		"workerSubnetCidr":        key.VnetWorkerSubnetCIDR(customObject),
+		"masterNodes":             masterNodes,
+		"workerNodes":             workerNodes,
+		"dnsZones":                key.DNSZones(customObject),
+		"hostClusterCidr":         r.azure.HostCluster.CIDR,
+		"kubernetesAPISecurePort": key.APISecurePort(customObject),
+		"masterCloudConfigData":   masterCloudConfig,
+		"workerCloudConfigData":   workerCloudConfig,
+		"vmssMSIEnabled":          r.azure.MSI.Enabled,
+		"templatesBaseURI":        baseTemplateURI(r.templateVersion),
```

Comment:
Why are those removed?

		"hostClusterResourceGroupName":  r.azure.HostCluster.ResourceGroup,
		"hostClusterVirtualNetworkName": r.azure.HostCluster.VirtualNetwork,

Reply:
done

I did also notice this does not belong in this PR